core/src/test/java/io/questdb/test/griffin/SqlCompilerImplTest.java

```diff
@@ -155,6 +152,21 @@ public void tesFailOnNonBooleanJoinCondition() throws Exception {
         );
     }
 
+    @Test
+    public void testCachedRecordCursorFactory() throws SqlException {
+        try (SqlCompilerImpl compilerX = new SqlCompilerImpl(engine) {
```

Comment:
`engine.compile()`

Reply:
how would I exercise the new interface which is only used in ent otherwise?